cpu/kinetis_common/uart.c

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (C) 2014 PHYTEC Messtechnik GmbH
+ * Copyright (C) 2014 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_kinetis_common_uart
+ *
+ * @{
+ *
+ * @file
+ * @brief       Low-level UART driver implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ *
+ * @}
+ */
+
+#include <math.h>
+
+#include "cpu.h"
+#include "thread.h"
+#include "sched.h"
+#include "periph_conf.h"
+#include "periph/uart.h"
+
+
+/**
+ * @brief Each UART device has to store two callbacks.
+ */
+typedef struct {
+    uart_rx_cb_t rx_cb;
+    uart_tx_cb_t tx_cb;
+    void *arg;
+} uart_conf_t;
+
+/**
+ * @brief Unified interrupt handler for all UART devices
+ *
+ * @param uartnum       the number of the UART that triggered the ISR
+ * @param uart          the UART device that triggered the ISR
+ */
+static inline void irq_handler(uart_t uartnum, KINETIS_UART *uart);
+
+/**
+ * @brief Allocate memory to store the callback functions.
+ */
+static uart_conf_t config[UART_NUMOF];
+
+static inline void kinetis_set_brfa(KINETIS_UART *dev, uint32_t baudrate, uint32_t clk)
+{
+#ifdef UART_C4_BRFA
+    /* set baudrate fine adjust (brfa) */
+    uint16_t brfa = ((((4 * clk) / baudrate) + 1) / 2) % 32;
+    dev->C4 = (uint8_t)UART_C4_BRFA(brfa);
```

Comment:
Is it necessary to cast to uint8_t here? Are there errors in any of the headers which need it?